server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -1954,21 +1955,16 @@ private boolean upgradeRunningVirtualMachine(Long vmId, Long newServiceOfferingI
     }
 
     @Override
-    public HashMap<String, VolumeStatsEntry> getVolumeStatistics(long clusterId, String poolUuid, StoragePoolType poolType, List<String> volumeLocators, int timeout) {
+    public HashMap<String, VolumeStatsEntry> getVolumeStatistics(long clusterId, String poolUuid, StoragePoolType poolType,  int timeout) {
         List<HostVO> neighbors = _resourceMgr.listHostsInClusterByStatus(clusterId, Status.Up);
         StoragePoolVO storagePool = _storagePoolDao.findPoolByUUID(poolUuid);
-        for (HostVO neighbor : neighbors) {
-            // apply filters:
-            // - managed storage
-            // - local storage
-            if (storagePool.isManaged() || storagePool.isLocal()) {
+        HashMap<String, VolumeStatsEntry> volumeStatsByUuid = new HashMap<>();
 
-                volumeLocators = getVolumesByHost(neighbor, storagePool);
-
-            }
+        for (HostVO neighbor : neighbors) {
+            List<String> volumeLocators = getVolumesByHost(neighbor, storagePool);
 
             // - zone wide storage for specific hypervisortypes
-            if (ScopeType.ZONE.equals(storagePool.getScope()) && storagePool.getHypervisor() != neighbor.getHypervisorType()) {
+            if ((ScopeType.ZONE.equals(storagePool.getScope()) && storagePool.getHypervisor() != neighbor.getHypervisorType()) || (volumeLocators == null || volumeLocators.size() == 0)) {
```

Comment:
@slavkap move this condition check `(ScopeType.ZONE.equals(storagePool.getScope()) && storagePool.getHypervisor() != neighbor.getHypervisorType()` before getting volume locators.

and use CollectionUtils for volumeLocators

```suggestion
            if (CollectionUtils.isEmpty(volumeLocators)) {
```

Reply:
Thanks @sureshanaparti ! I'll do the change suggestion, but may I ask is it appropriate if I push this PR in earlier version of CS for example 4.13.2.0?

Reply:
@slavkap yes, we will then merge forward, no problem (except maybe a conflict to resolve)

Reply:
Thanks @DaanHoogland! I'll fix the conflicts and will push it against 4.13.2.0

Reply:
may I ask for advice, because I'm not sure how to proceed? I've never changed the branch of a PR, so what is the right action - to edit this PR branch or to checkout to a branch which is on top of 4.13, to cherry-pick the changes (resolve conflicts) and to create new pull request?

Reply:
you can try and rebase the range of your commit on top of 4.13, or you can cherrypinck. when done you should force push and you can then change the base branch.

Reply:
thanks again @DaanHoogland !